neuronunit/models/backends.py

```diff
@@ -423,28 +402,127 @@ def inject_square_current(self, current):
         self.h(prefix+'amplitude=%s'%amps)
         self.h(prefix+'duration=%s'%c['duration'])
         self.h(prefix+'delay=%s'%c['delay'])
-        self.local_run()
 
+        ##
+        # Duplicated call to run?
+        self.local_run()
```

Comment:
@russelljjarvis 
How are you able to call `local_run()` here but not inside of `run()`?  I don't like the idea of `local_run` being called in `set_attrs` which should really be about setting attributes rather than running.

Reply:
local_run() is not currently called in `set_attrs` in model.__init__.py, it is called in NEURONbackend `inject_square_current` however.

I do not know why that line brakes, the code.

Reply:
See above.  I think `local_run()` should be called nowhere except within `run()`.  Everything else is just configuring the model to be run.